index.bs

```diff
@@ -324,10 +339,11 @@ dictionary NavigatorUABrandVersion {
 };
 
 dictionary UADataValues {
-  DOMString platform; 
-  DOMString platformVersion;
-  DOMString architecture;
-  DOMString model;
+  DOMString platform = ""; 
```

Comment:
@domenic - brought back empty string initialization to this returned dictionary, as otherwise, the tests fail (as the non-requested values became `undefined` instead of being an empty strring)

Reply:
That must be a Chromium implementation issue? Defaults do not work that way per spec; they only apply when used as an incoming argument. You need to remove it from here and add explicit steps to set them to empty string in the algorithm, if that's the desired behavior.

However, that behavior seems wrong (i.e. the tests seem wrong). Any non-requested fields are way better as being actually not present. Why have them be present with a misleading empty string value?

Reply:
Fair enough regarding the default values and I can change the test to look for `undefined`.

Should I also file an issue on Chromium's IDL implementation? 
Looking at https://heycam.github.io/webidl/#dfn-dictionary-member-default-value, can you point me to the specific part that indicates this behavior is wrong?

Reply:
Even better would be to have the test check for not present, e.g. `!("platform" in returnValue)`.

https://heycam.github.io/webidl/#es-dictionary "An IDL dictionary value V is converted to an ECMAScript Object value by running the following algorithm (where D is the dictionary):" is the algorithm that's supposed to be implemented here. (Your algorithm is returning an IDL dictionary, which gets converted into an ECMAScript object for exposure to JS consumers.) Step 3.1.2 shows how the ES object only gets a property if the dictionary member is present. If it's not present, then the default value is not consulted; instead, no property is defined.

So yeah, filing a Chromium bug would probably be good :).

Reply:
Filed https://bugs.chromium.org/p/chromium/issues/detail?id=1059695 and reverted the initialization